### PR TITLE
OXY-76-publish loan repayment reminders to ActiveMQ

### PR DESCRIPTION
**Pull request by hero78**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java [hunk 1]
```diff
@@ -20,9 +20,11 @@
```
Comment (hiteshprajapati) [inline fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java]: @hero78 , As per our talk, I wouldn't expect change anything in this class.

**Review by hiteshprajapati — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfigurationForGeneralActiveMq.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    @Bean
+    public Logger loggerBean() {
+        return LoggerFactory.getLogger(MessagingConfigurationForGeneralActiveMq.class);
+    }
+
+    @Bean
+    public ActiveMQConnectionFactory amqConnectionFactory() {
+        ActiveMQConnectionFactory amqConnectionFactory = new ActiveMQConnectionFactory(); // NOSONAR
+        try {
+            amqConnectionFactory.setBrokerURL(this.env.getProperty("ACTIVEMQ_BLOCKER_URL"));
```
Comment (hiteshprajapati) [inline fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfigurationForGeneralActiveMq.java]: you need to fetch property using property name. the propery name in application.properties is not ACTIVEMQ_BLOCKER_URL , but its fineract.activemq.blockerUrl
so it should be this.env.getProperty("fineract.activemq.blockerUrl"); ?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanSchedularServiceImpl.java [hunk 3]
```diff
@@ -355,4 +396,26 @@ private AppUser getAppUserIfPresent() {
...
+    private void buildNotification(String permission, String objectType, Long objectIdentifier, String notificationContent,
+            String eventType, Long appUserId, Long officeId) {
+
+        String tenantIdentifier = ThreadLocalContextUtil.getTenant().getTenantIdentifier();
+        Set<Long> userIds = getNotifiableUserIds(officeId, permission);
+        NotificationData notificationData = new NotificationData(objectType, objectIdentifier, eventType, appUserId, notificationContent,
+                false, false, tenantIdentifier, officeId, userIds);
+        try {
+            notificationEventPublisher.broadcastNotificationLoanRepaymentReminders(notificationData,
+                    this.env.getProperty("LOAN_REPAYMENT_REMINDER_QUEUE"));
```
Comment (hiteshprajapati) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanSchedularServiceImpl.java]: fetch property with 
 this.env.getProperty("fineract.activemq.loanRepaymentReminderQueue");?

**Comment by Napho** [comment]
@hero78  dont forget to squash related commits

File: fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfigurationForGeneralActiveMq.java (hunk 2 above)
Comment (hero78) [reply fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfigurationForGeneralActiveMq.java]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanSchedularServiceImpl.java (hunk 3 above)
Comment (hero78) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanSchedularServiceImpl.java]: fixed

File: fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java (hunk 1 above)
Comment (hero78) [reply fineract-provider/src/main/java/org/apache/fineract/notification/config/MessagingConfiguration.java]: fixed

**Review by hiteshprajapati — APPROVED** [review]

**Review by Deepika1095 — APPROVED** [review]

**Pull request merged**